File: src/main/java/org/profit/app/schedule/StockExecutor.java
package org.profit.app.schedule;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

/**
 * 线程池
 *
 * @author TangYing
 */
public enum StockExecutor {

	INSTANCE;

	private final Logger LOG = LoggerFactory.getLogger(StockExecutor.class);

	private ExecutorService service;

	public void start() {
		service = Executors.newFixedThreadPool(4);
		LOG.info("StockExecutor is started.");
	}

	public void shutdown() {
		service.shutdown();
	}
}

File: src/main/java/org/profit/app/schedule/StockExecutor.kt
package org.profit.app.schedule

import org.profit.app.service.DownloadHistoryService
import org.slf4j.LoggerFactory
import java.util.concurrent.ExecutorService
import java.util.concurrent.Executors

/**
 * 线程池
 */
object StockExecutor {

    private val service: ExecutorService = Executors.newFixedThreadPool(8)

    fun shutdown() {
        service.shutdown()
    }

    fun download(code: String, date: String) {
        service.submit{
            println("$code 开始下载 $code 的数据...")
            DownloadHistoryService(code, date).execute()
        }
    }
}